test/e2e/25_max_history.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+function setup() {
+  # Load libraries in setup() to access BATS_* variables 
+  load lib/env
+  load lib/install
+  load lib/poll
+
+  kubectl create namespace "$E2E_NAMESPACE"
+  install_git_srv
+  install_tiller
+  install_helm_operator_with_helm
+  kubectl create namespace "$DEMO_NAMESPACE"
+}
+
+@test "When max history is not set on release the most recent 10 revisions are kept" {
+  if [ "$HELM_VERSION" != "v3" ]; then
+    skip
+  fi
+
+  # Apply the HelmRelease
+  kubectl apply -f "$FIXTURES_DIR/releases/helm-repository.yaml" >&3
+  poll_until_equals 'podinfo-helm-repository HelmRelease' 'deployed' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.releaseStatus' --no-headers"
+
+  for i in {1..15}
+  do
+    # Apply a patch to initiate an upgrade
+    kubectl patch -f "$FIXTURES_DIR/releases/helm-repository.yaml" --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/values/someField\", \"value\":\"$i\"}]" >&3
+    poll_until_equals 'podinfo-helm-repository HelmRelease' "$(($i+1))" "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o 'custom-columns=status:status.observedGeneration' --no-headers"
+  done
+
+  # Check count of revisions is <= 10
+  count=$(k get secret -n $DEMO_NAMESPACE --output name | grep 'sh.helm.release.v1.podinfo-helm-repository' | wc -l)
+  [ "$count" -le 10 ]
```

Comment:
I think it is wiser to not use our human knowledge here about the configured storage mechanism but instead `exec` into the running container and use the `helm3` binary there to get this information. This would make the end-to-end tests still succeed if for some reason the storage mechanism is configured differently.

I also think the stricter `[ "$count" -eq 10 ]` can be used, given we use `poll_until_equals` in the `for` loop and there should thus be no divergence due to e.g. the operator not being ready with processing.